pkg/executor/show_stats.go

```diff
@@ -114,36 +118,53 @@ func (e *ShowExec) fetchShowStatsMeta(ctx context.Context) error {
 	do := domain.GetDomain(e.Ctx())
 	h := do.StatsHandle()
 	dbs := do.InfoSchema().AllSchemaNames()
+	var (
+		fieldPatternsLike collate.WildcardPattern
+		fieldFilter       string
+	)
+	if e.Extractor != nil {
+		fieldFilter = e.Extractor.Field()
+		fieldPatternsLike = e.Extractor.FieldPatternLike()
+	}
+	tableInfoResult := do.InfoSchema().ListTablesWithSpecialAttribute(infoschemacontext.PartitionAttribute)
+	paritionedTables := make(map[int64]*model.TableInfo)
+	for _, result := range tableInfoResult {
+		for _, tbl := range result.TableInfos {
+			paritionedTables[tbl.ID] = tbl
+		}
+	}
 	for _, db := range dbs {
-		tables, err := do.InfoSchema().SchemaTableInfos(ctx, db)
-		if err != nil {
-			return errors.Trace(err)
+		if fieldFilter != "" && db.L != fieldFilter {
+			continue
+		} else if fieldPatternsLike != nil && !fieldPatternsLike.DoMatch(db.L) {
+			continue
 		}
-		for _, tbl := range tables {
-			pi := tbl.GetPartitionInfo()
-			if pi == nil || e.Ctx().GetSessionVars().IsDynamicPartitionPruneEnabled() {
-				partitionName := ""
-				if pi != nil {
-					partitionName = "global"
-				}
-				e.appendTableForStatsMeta(db.O, tbl.Name.O, partitionName, h.GetTableStats(tbl))
-				if pi != nil {
-					for _, def := range pi.Definitions {
-						e.appendTableForStatsMeta(db.O, tbl.Name.O, def.Name.O, h.GetPartitionStats(tbl, def.ID))
-					}
+		tableNames, err := do.InfoSchema().SchemaSimpleTableInfos(ctx, db)
+		terror.Log(err)
+		for _, nameInfo := range tableNames {
+			tblID := nameInfo.ID
+			paritionedTable, ok := paritionedTables[tblID]
```

Comment:
I find that,  in theory,  the speical attribute tables in infoschema V2 can provide TableInfoByID ability.
And if we do provide such API, we can avoid calling `ListTablesWithSpecialAttribute` to making the temporary hash table here. 

Never mind, since we did provoid such API, and its unrelated to this PR, just go ahead.

Reply:
> I find that, in theory, the speical attribute tables in infoschema V2 can provide TableInfoByID ability.

Do you mean only for the partitioned tables or all tables? Or are you implying that we can use TableInfoByID instead of the special API here?

Reply:
For partition tables only.
I mean if we provide a TableInfoByID just for partition tables ... that kind of special API could help here.